ui/apps/ui/src/app/components/filters/filter-multiselect/first-n-values.component.ts

```diff
@@ -45,11 +45,27 @@ export class FirstNValuesComponent implements OnChanges {
   @Output()
   toggleActive = new EventEmitter<[IUIFilterTreeNode, boolean][]>();
 
+  @Output()
+  hasSelectedFiltersOutput = new EventEmitter<boolean>();
+
   ngOnChanges(changes: SimpleChanges) {
     if (changes['query'] || changes['allEntities'] || changes['displayMax']) {
       this._allEntities = flatNodesToTree(
         search(this.query, this.allEntities)
       ).slice(0, this.displayMax);
     }
+
+    if (changes['allEntities'].currentValue) {
+      // Mapping so we don't get TS errors
+      const currentValues: { isSelected: boolean }[] =
+        changes['allEntities'].currentValue;
+
+      // Workaround for Error: NG0100: ExpressionChangedAfterItHasBeenCheckedError
+      Promise.resolve().then(() => {
+        this.hasSelectedFiltersOutput.emit(
+          currentValues.some(({ isSelected }) => isSelected)
+        );
+      });
+    }
```

Comment:
I'd like some advice on a better way of handling this perhaps?

Reply:
**import { ChangeDetectionStrategy } from '@angular/core';**
@Component({
  selector: 'ess-first-n-values',
  **changeDetection: ChangeDetectionStrategy.OnPush**
  template: ` <div class="filter__viewport">
    <ess-checkboxes-tree
      [data]="_allEntities"
      (checkboxesChange)="toggleActive.emit($event)"
    ></ess-checkboxes-tree>
  </div>`,
  styles: [
    `
      .filter__viewport {
        max-height: 290px;
        overflow: auto;
      }
    `,
  ],
})

Reply:
I've changed the logic of letting the filters know that there aren't any active. Now I've added some logic to the _customRoute observables in the multiselect component. This should be clearer